### PR TITLE
dotnet rule updates

### PR DESCRIPTION
**Pull request by mike-hunhoff**
adding new dotnet rules and updating existing rules with dotnet APIs.

### CODE REVIEW

File: communication/http/client/send-http-request.yml [hunk 1]
```diff
@@ -12,6 +13,7 @@ rule:
       - 6A352C3E55E8AE5ED39DC1BE7FB964B1:0x100026E0
   features:
     - or:
+      - api: System.Net.WebRequest::GetResponse
```
Comment (mr-tz) [inline communication/http/client/send-http-request.yml:16]: shouldn't it be `Create` or similar here?

File: nursery/mixed-mode-dotnet.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+rule:
+  meta:
+    name: mixed mode dotnet
```
Comment (mr-tz) [inline nursery/mixed-mode-dotnet.yml:3]: would be helpful to add a description of what this means

File: nursery/unmanaged-call-dotnet.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+rule:
+  meta:
+    name: unmanaged call dotnet
```
Comment (mr-tz) [inline nursery/unmanaged-call-dotnet.yml:3]: would be helpful to add a description of what this means

**Review by mr-tz — APPROVED** [review]
nice, added a few minor comments

File: nursery/unmanaged-call-dotnet.yml (hunk 3 above)
Comment (williballenthin) [inline nursery/unmanaged-call-dotnet.yml:3]:
```suggestion
    name: unmanaged .NET call
```

**Comment by williballenthin** [comment]
ping @mike-hunhoff for comments above by moritz

File: nursery/unmanaged-call-dotnet.yml (hunk 3 above)
Comment (mike-hunhoff) [reply nursery/unmanaged-call-dotnet.yml:3]: updated w/ description

Comment (mike-hunhoff) [reply nursery/unmanaged-call-dotnet.yml:3]: We may want to make this more generic to open up the rule to other runtimes that may be supported in the future.

File: nursery/mixed-mode-dotnet.yml (hunk 2 above)
Comment (mike-hunhoff) [reply nursery/mixed-mode-dotnet.yml:3]: added

[the remark this answers is not in the document]
Comment (mike-hunhoff) [reply nursery/mixed-mode-dotnet.yml:3]: We may want to make this more generic to open up the rule to other runtimes that may be supported in the future.

**Pull request merged**